utilities/shared_disk.py

```
@@ -253,58 +291,587 @@ def verify_shared_disk_data(
+            break
+    if not backing_uuid:
+        raise ValueError(f"Cannot find backing UUID for shared VMDK '{shared_vmdk_path}' on VM '{owner_name}'")
+
+    serial = backing_uuid.replace("-", "").lower()
+    LOGGER.info(f"Shared disk on '{owner_name}': backing.uuid={backing_uuid}, serial={serial}")
+    return serial
+
+
+def _execute_guest_label_command(
+    source_provider: "VMWareProvider",
+    owner_vm: Any,
```

Ruff (0.15.18)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `owner_vm`

(ANN401)